### PR TITLE
chore: Templates UI updates

### LINKED ISSUE #11794: [Bug]: The dropdown alignment is broken inside fork modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The dropdown alignment is broken inside fork modal
image

### Steps To Reproduce

1. Inside the Home page
2. click on meat ball icon below the app
3. Click on fork
4. Indie the fork modal click on the org name dropdown
5. The alignment is broken 

### Public Sample App

_No response_

### Version

release

### LINKED ISSUE #12270: [Task]: Templates UI tweaks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [x] Remove usecases and function from filters for now
- [x] UI changes
- [x] Apply OR logic to filters previously it was OR.
- [x] Expand all filters for now.
- [x] Go back should work like browser back button

figma: https://www.figma.com/file/NV3FgZqqe88mU4Tozmwhmd/Templates-Phase-1-Designs-Handoff?node-id=2%3A1

_No response_

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

UI tweaks in templates
fork icon has been changes for fork template and fork application
Make use of the api to get the template by id. 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

Fixes #12270 
Fixes #11794

## How Has This Been Tested?

Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/minor-ui-fixes 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.68 **(-0.01)** | 37.05 **(0)** | 35.23 **(-0.02)** | 55.94 **(0)**
 :green_circle: | app/client/src/actions/templateActions.ts | 68 **(-0.18)** | 100 **(0)** | 0 **(0)** | 90 **(1.11)**
 :red_circle: | app/client/src/ce/constants/messages.ts | 76.92 **(-0.03)** | 100 **(0)** | 31.59 **(-0.09)** | 81.87 **(0.04)**
 :red_circle: | app/client/src/reducers/uiReducers/templateReducer.ts | 29.41 **(-6.3)** | 100 **(0)** | 0 **(0)** | 29.41 **(-6.3)**
 :red_circle: | app/client/src/sagas/TemplatesSagas.ts | 23.08 **(-3.59)** | 5 **(-0.56)** | 12.5 **(-1.79)** | 28.57 **(-3.86)**</details>

### CODE REVIEW

File: app/client/src/pages/Templates/index.tsx [hunk 1]
```diff
@@ -37,13 +37,15 @@ const PageWrapper = styled.div`
...
   height: calc(100vh - ${(props) => props.theme.homePage.header}px);
+  padding-left: 9vw;
 `;
 
 export const TemplateListWrapper = styled.div`
   padding-top: ${(props) => props.theme.spaces[11]}px;
   width: calc(100% - ${(props) => props.theme.homePage.sidebar}px);
   height: calc(100vh - ${(props) => props.theme.headerHeight});
   overflow: auto;
+  padding-right: 9vw;
```
Comment (arunvjn) [inline app/client/src/pages/Templates/index.tsx:48]: Any reason for using 'vw' here ?

File: app/client/src/pages/Templates/index.tsx [hunk 2]
```diff
@@ -37,13 +37,15 @@ const PageWrapper = styled.div`
   margin-top: ${(props) => props.theme.homePage.header}px;
   display: flex;
   height: calc(100vh - ${(props) => props.theme.homePage.header}px);
+  padding-left: 9vw;
```
Comment (arunvjn) [inline app/client/src/pages/Templates/index.tsx:40]: Any reason for using 'vw' here ?

File: app/client/src/pages/Templates/index.tsx (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/pages/Templates/index.tsx:48]: Wanted a bit more space for larger width screens as in want it to be relative to the view port width.

File: app/client/src/globalStyles/popover.ts [hunk 3]
```diff
@@ -23,6 +23,10 @@ export const PopoverStyles = createGlobalStyle`
   .onboarding-carousel .${Classes.OVERLAY_CONTENT} {
     filter: drop-shadow(0px 6px 20px rgba(0, 0, 0, 0.15));
   }
+  .templates-notification .bp3-popover2-arrow {
+    // !important because top is specified as an inline style in the lib
+    top: -9px !important;
```
Comment (albinAppsmith) [inline app/client/src/globalStyles/popover.ts:28]: Can we use multiples of 4 here?

File: app/client/src/pages/Templates/Template/RequestTemplate.tsx [hunk 4]
```diff
@@ -43,7 +43,7 @@ const Wrapper = styled.div`
 `;
 
 const StyledImage = styled.img`
-  height: 147px;
+  height: 170px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Templates/Template/RequestTemplate.tsx:46]: Make it multiples of 4

File: app/client/src/pages/Templates/Template/index.tsx [hunk 5]
```diff
@@ -26,14 +26,15 @@ const TemplateWrapper = styled.div`
...
+  padding: ${(props) => props.theme.spaces[9]}px;
   overflow: hidden;
 `;
 
 const StyledImage = styled.img`
   box-shadow: 0px 17.52px 24.82px rgba(0, 0, 0, 0.09);
-  object-fit: cover;
+  object-fit: contain;
+  width: 100%;
+  height: 235px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Templates/Template/index.tsx:37]: Make it multiples of 4

File: app/client/src/pages/Templates/TemplatesTabItem.tsx [hunk 6]
```diff
@@ -24,14 +24,20 @@ import { AppState } from "reducers";
...
+  .${Classes.ICON} {
+    align-items: unset;
+    margin-top: ${(props) => props.theme.spaces[0] + 1}px;
+  }
 
   .text-wrapper {
     display: flex;
     flex-direction: column;
-    margin-left: ${(props) => props.theme.spaces[8]}px;
+    margin-left: ${(props) => props.theme.spaces[4]}px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Templates/TemplatesTabItem.tsx:40]: Can we make it 8 or 12?

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**